### PR TITLE
Drop `decoder_input_ids` in `DPOTrainer`

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Since we only use the logits (and not the returned loss), it seems like `decoder_input_ids` is useless as an input for encoder-decoder models.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qgallouedec** [comment]
The new curves are significantly smoother compared to before. 

I believe this is due to padding completions without applying a loss mask, causing the loss to be calculated over the entire sequence, including the padding tokens:  
https://github.com/huggingface/trl/blob/7e5924d17ebf7036f03091d60bde15e2367e7fe6/trl/trainer/dpo_trainer.py#L278-L283

**Review by kashif — APPROVED** [review]
